src/App.jsx

```diff
@@ -1,15 +1,60 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import messages from './data/messages.json';
 
-const App = () => {
+
+const calculateTotalLikeCount = (chat) => {
+  return chat.reduce((total, chat) => {
+    return total + (chat.liked ? 1 : 0);
+  }, 0);
+};
+
+const localSender = messages[0].sender;
+let remoteSender = 'Unknown';
+
+for (const message of messages) {
+  if (message.sender !== localSender) {
+    remoteSender = message.sender;
+    break; // Stop once other sender is found
+  }
+}
+
+
+const App =() => {
+  const [chat, setChat] = useState(messages);
+
+  const handleLikeButtom = (id) => {
+    setChat(chat => {
+      return chat.map(chat => {
+        if (chat.id === id) {
+          return { ...chat, liked: !chat.liked };
+        } else {
+          return chat;
+        }
+      });
+    });
+  };
```

Comment:
No notes on this! Shows me that you have a great understanding of lifting up state!